src/modules/AutoinstClass.rb

```diff
@@ -276,10 +234,10 @@ def compat_class_file
     # Builds a map of classes to import from /etc/autoinstall/classes.xml
     # @return [Array<Hash>] Classes defined in the file.
     def read_old_classes
-      old_classes_map = Convert.to_map(SCR.Read(path('.xml'), compat_class_file))
-      old_classes = old_classes_map['classes'] || []
+      old_classes_map = Convert.to_map(SCR.Read(path(".xml"), compat_class_file))
+      old_classes = old_classes_map["classes"] || []
```

Comment:
This method is called only if `/etc/autoinstall/classes.xml` exists but you're right: if the file is empty or contains invalid XML, `old_classes_map` will be nil.

I've added a new test to cover this case.